typescript/sdk/src/providers/MultiProtocolProvider.test.ts

```diff
@@ -0,0 +1,26 @@
+import { expect } from 'chai';
+
+import { ethereum } from '../consts/chainMetadata';
+import { Chains } from '../consts/chains';
+import { MultiProtocolProvider } from '../providers/MultiProtocolProvider';
+
+describe('MultiProtocolProvider', () => {
+  describe('constructs', () => {
+    it('creates a multi protocol provider without type extension', async () => {
+      const multiProvider = new MultiProtocolProvider();
+      const ethMetadata = multiProvider.getChainMetadata(Chains.ethereum);
+      expect(ethMetadata.name).to.equal(Chains.ethereum);
+    });
+    it('creates a multi protocol provider with type extension', async () => {
+      const multiProvider = new MultiProtocolProvider<{
+        ism: string;
+        count: number;
+      }>({
+        [Chains.ethereum]: { ...ethereum, ism: '0x123', count: 1 },
```

Comment:
im a bit confused what's going on here
ism config to the multi provider?

Reply:
Sorry, poor choice of name on my part. It was meant to be a random property just to test that the generic typing is working right. I'll rename